pkg/frontend/clustermanager_delete.go

```diff
@@ -0,0 +1,56 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+)
+
+func (f *frontend) deleteClusterManagerConfiguration(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+
+	if disableOCMAPI {
+		reply(log, w, nil, []byte("forbidden."), api.NewCloudError(http.StatusForbidden, api.CloudErrorCodeForbidden, "", "forbidden."))
+		return
+	}
+
+	err := f._deleteClusterManagerConfiguration(ctx, log, r)
+	switch {
+	case cosmosdb.IsErrorStatusCode(err, http.StatusNotFound):
+		err = statusCodeError(http.StatusNoContent)
+	case err == nil:
+		err = statusCodeError(http.StatusAccepted)
+	}
+	reply(log, w, nil, nil, err)
+}
+
+func (f *frontend) _deleteClusterManagerConfiguration(ctx context.Context, log *logrus.Entry, r *http.Request) error {
+	vars := mux.Vars(r)
+
+	doc, err := f.dbClusterManagerConfiguration.Get(ctx, r.URL.Path)
+	switch {
+	case cosmosdb.IsErrorStatusCode(err, http.StatusNotFound):
+		return api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeResourceNotFound, "", "The Resource '%s/%s' under resource group '%s' was not found.", vars["resourceType"], vars["resourceName"], vars["resourceGroupName"])
+	case err != nil:
+		return err
+	}
+
+	doc.Deleting = true
+	_, err = f.dbClusterManagerConfiguration.Update(ctx, doc)
+	if err != nil {
+		return err
+	}
+
+	err = f.dbClusterManagerConfiguration.Delete(ctx, doc)
+	return err
```

Comment:
I will remove the deletion code, but we wont be able to reuse the same resource since we will get stale documents since there is nothing else cleaning them up

Reply:
@bennerv I actually have a [commit on my M7 branch](https://github.com/Azure/ARO-RP/pull/2363/commits/cf42c900af5c5a3cc1564c1c60565e1915ed25cc) to change this.  Maybe it should be cherry-picked in?

I talked Ross into having the _frontend_ set a TTL on the document rather than deleting directly as a last-minute hack, since the backend at present only reads the change feed and doesn't interact with the database directly.  I realized pretty late that document deletion erases all traces of the document from the change feed, which surprised me.